lightningd/peer_control.c

```diff
@@ -617,6 +617,21 @@ static void
 peer_connected_serialize(struct peer_connected_hook_payload *payload,
 			 struct json_stream *stream)
 {
+	const struct peer *p = payload->peer;
+	const struct channel *channel = payload->channel;
+	json_object_start(stream, "peer");
+	json_add_pubkey(stream, "id", &p->id);
+	json_add_string(
+	    stream, "addr",
+	    type_to_string(stream, struct wireaddr_internal, &payload->addr));
+	json_add_hex_talarr(stream, "globalfeatures", p->globalfeatures);
+
+	if (channel != NULL) {
+		json_add_channel(payload->ld, stream, "channel", channel);
+	} else {
+		json_add_null(stream, "channel");
+	}
+	json_object_end(stream); /* .peer */
```

Comment:
I think this hook may be far too helpful.  Just hand it the peer id, address and the global and local features from this connection message; the rest it can get via an RPC call if it wants?